railties/lib/rails/application/default_middleware_stack.rb

```diff
@@ -16,6 +16,11 @@ def build_stack
           middleware.use ::ActionDispatch::HostAuthorization, config.hosts, config.action_dispatch.hosts_response_app
 
           if config.force_ssl
+            if config.action_dispatch.ssl_default_redirect_status && !config.ssl_options.dig(:redirect, :status)
+              config.ssl_options[:redirect] ||= {}
+              config.ssl_options[:redirect][:status] = config.action_dispatch.ssl_default_redirect_status
+            end
+
             middleware.use ::ActionDispatch::SSL, **config.ssl_options
```

Comment:
It's a little strange that we're mutating the application configuration here. How about passing `ssl_default_redirect_status` through to the middleware instead, e.g.:

```suggestion
            middleware.use ::ActionDispatch::SSL, **config.ssl_options,
              ssl_default_redirect_status: ssl_default_redirect_status
```

```ruby
      def redirection_status(request)
        if ssl_default_redirect_status
          ssl_default_redirect_status
        elsif request.get? || request.head?
          301 # Issue a permanent redirect via a GET request.
        else
          307 # Issue a fresh request redirect to preserve the HTTP method.
        end
      end
```